contrib/w32mousehook/com/sun/jna/contrib/demo/MouseHook.java

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2020 Dmitry Prokhorov, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+
+package com.sun.jna.contrib.demo;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.platform.win32.Kernel32;
+import com.sun.jna.platform.win32.User32;
+import com.sun.jna.platform.win32.WinDef;
+import com.sun.jna.platform.win32.WinUser;
+
+public class MouseHook {
+    private static WinUser.HHOOK hook;
+
+    public static void main(String[] args) {
+        hook = User32.INSTANCE.SetWindowsHookEx(WinUser.WH_MOUSE_LL, new WinUser.LowLevelMouseProc() {
```

Comment:
This is invalid. The anonymous class `WinUser.LowLevelMouseProc` becomes immediately eligible for garbadge collection. You need to keep a strong reference on the java side. In the keyboard hook the callback is stored in a static field, which is one option.